src/io/fabric8/Fabric8Commands.groovy

```diff
@@ -150,23 +150,50 @@ def getNewVersionFromTag(){
     return version
   }
 
-  tag = tag.trim()
-
-  // strip the v prefix from the tag so we can use in a maven version number
-  def previousReleaseVersion = tag.substring(tag.lastIndexOf('v')+1)
-  echo "Previous version found ${previousReleaseVersion}"
+  def semver = tag =~ /\bv?(?<major>0|[1-9]\d*)(?:\.(?<minor>0|[1-9]\d*)(?:\.(?<patch>0|[1-9]\d*))?)?(?:-(?<prerelease>[\da-z\-]+(?:\.[\da-z\-]+)*))?(?:\+(?<build>[\da-z\-]+(?:\.[\da-z\-]+)*))?\b/
+
+  if (semver.matches()) {
+    def majorVersion = semver.group('major') as int
+    def minorVersion = (semver.group('minor') ?: 0) as int
+    def patchVersion = ((semver.group('patch') ?: 0) as int) + 1
+
+    def pomSemver = pomVersion =~ /\bv?(?<major>0|[1-9]\d*)(?:\.(?<minor>0|[1-9]\d*)(?:\.(?<patch>0|[1-9]\d*))?)?(?:-(?<prerelease>[\da-z\-]+(?:\.[\da-z\-]+)*))?(?:\+(?<build>[\da-z\-]+(?:\.[\da-z\-]+)*))?\b/
+    if (pomSemver.matches) {
+      def pomMajorVersion = pomSemver.group('major') as int
+      def pomMinorVersion = (pomSemver.group('minor') ?: 0) as int
+      def pomPatchVersion = (pomSemver.group('patch') ?: 0) as int
+
+      if (pomMajorVersion > majorVersion) {
+        majorVersion = pomMajorVersion
+        minorVersion = pomMinorVersion
+        patchVersion = pomPatchVersion
+      } else if (pomMajorVersion == majorVersion && pomMinorVersion > minorVersion) {
+        minorVersion = pomMinorVersion
+        patchVersion = pomPatchVersion
+      } else if (pomMajorVersion == majorVersion && pomMinorVersion == minorVersion && pomPatchVersion > patchVersion) {
+        patchVersion = pomPatchVersion
+      }
```

Comment:
You could put all into one if since resetting the version with the same value does not harm:

```java
if (pomMajorVersion > majorVersion || 
    (pomMajorVersions == majorVersion && pomMinorVersion > minorVersion) ||
    (pomMajorVersions == majorVersion && pomMinorVersion == minorVersion && pomPatchVersion > patchVersion)) {
  majorVersion = pomMajorVersion
  minorVersion = pomMinorVersion
  patchVersion = pomPatchVersion
} 
```
assuming the a version can never decrease (i.e. that the tag major version is never higher than the one provided.)

On the other hand I wonder why not relying only on the `pomVersion` ? E.g. we could have a property "releaseVersion" which always holds the last release version. CD would simply increment that. If one wants to change the one of the major minor versions one could do it manually. If one want to starts a new level, one could specify "1.4.-1", CD would do the jump to "1.4.0" then. (Or could be also "1.4" as indication for a new round of patch levels).

I like to have things in one place so that people don't have to jump between systems (Maven, Git) for doing a task like setting the release number.